core/src/engine/coro/pool.hpp

```diff
@@ -157,6 +207,26 @@ PoolStats Pool<Task>::GetStats() const {
   return stats;
 }
 
+template <typename Task>
+void Pool<Task>::ClearLocalCache() {
+  const std::size_t current_idle_coroutines_num =
+      idle_coroutines_num_.load(std::memory_order_acquire);
+  if (current_idle_coroutines_num < config_.max_size) {
```

Comment:
This `if` can be transformed into making `return_to_pool_from_local_cache_num = 0` if there is no need to return any coroutines. Perhaps use `std::clamp`? The code may become a little bit simpler

Reply:
I don't know how it's easier using std::clamp. Now it seems to me that the simplest condition is. Provided that due to race, current_idle_coroutines_num may exceed config_.max_size.
This race has happened before, does it need to be corrected? This will require adding cas and will likely slow down.

Reply:
OK, no need to guard against occasionally exceeding `max_size`